substra/sdk/client.py

```diff
@@ -44,132 +47,92 @@ def wrapper(*args, **kwargs):
+        self._backend = self._get_backend()
 
-        if token:
-            self._set_token(token)
+    def _get_backend(self):
+        return backends.get(
+            self._backend_name,
+            url=self._url,
+            version=self._version,
+            insecure=self._insecure,
+            token=self._token,
+            retry_timeout=self._retry_timeout,
+        )
 
     @logit
     def login(self, username, password):
         """Login to a remote server. """
-
-        if not self._current_profile:
-            raise exceptions.SDKException("No profile defined")
-
-        res = self.client.login(username, password)
-        token = res.json()['token']
-        self._set_token(token)
-        return token
-
-    def _set_token(self, token):
-        if not self._current_profile:
-            raise exceptions.SDKException("No profile defined")
-
-        self._current_profile.update({
-            'token': token,
-        })
-        self.client.set_config(self._current_profile, self._profile_name)
-        return token
-
-    def _set_current_profile(self, profile_name, profile):
-        """Set client current profile."""
-        self._profile_name = profile_name
-        self._profiles[profile_name] = profile
-        self._current_profile = profile
-        self.client.set_config(self._current_profile, profile_name)
-
-        return profile
-
-    def set_profile(self, profile_name):
-        """Set profile from profile name.
-
-        If profiles has not been defined through the `add_profile` method, it is loaded
-        from the config file.
-        """
-        try:
-            profile = self._profiles[profile_name]
-        except KeyError:
-            profile = self._cfg_manager.load_profile(profile_name)
-
-        return self._set_current_profile(profile_name, profile)
-
-    def set_user(self):
-        try:
-            user = self._usr_manager.load_user()
-        except (exceptions.UserException, FileNotFoundError):
-            pass
-        else:
-            if self._current_profile is not None and 'token' in user:
-                self._current_profile.update({
-                    'token': user['token'],
-                })
-                self.client.set_config(self._current_profile, self._profile_name)
-
-    def add_profile(self, profile_name, url, version='0.0', insecure=False):
-        """Add new profile (in-memory only)."""
-        profile = cfg.create_profile(
-            url=url,
-            version=version,
-            insecure=insecure,
+        if not self._backend:
+            raise exceptions.SDKException("No backend found")
+        self._token = self._backend.login(username, password)
+        return self._token
+
+    @staticmethod
+    def _get_spec(asset_type, data):
+        if isinstance(data, asset_type):
+            return data
+        return asset_type(**data)
+
+    @classmethod
+    def from_config_file(
+        cls,
+        profile_name: str,
```

Comment:
Weirdly enough, I think profile_name should be optional as well, since there is a default value.